Process.c

```diff
@@ -1073,6 +1079,51 @@ void Process_updateCPUFieldWidths(float percentage) {
    Row_updateFieldWidth(PERCENT_NORM_CPU, width);
 }
 
+void Process_updateCPUGroupFieldWidth(float percentage) {
+   if (percentage < 99.9F) {
+      Row_updateFieldWidth(PERCENT_CPU_GROUP, 4);
+      return;
+   }
+
+   // Add additional two characters, one for "." and another for precision.
+   uint8_t width = ceil(log10(percentage + 0.1)) + 2;
+
+   Row_updateFieldWidth(PERCENT_CPU_GROUP, width);
+}
+
+void Process_resetGroupFields(pid_t _pid, Process* this, void* _data) {
+   this->percent_cpu_group = this->percent_cpu;
+   this->percent_mem_group = this->percent_mem;
+}
+
+void Process_updateGroupFields(pid_t _pid, Process* this, void* _data) {
+   const Machine* host = this->super.host;
+   const ProcessTable* pt = (const ProcessTable*) host->processTable;
+   Process* curProcess = this;
+
+   bool looping = curProcess->percent_cpu_group > 0 || curProcess->percent_mem_group > 0;
+
+   while (looping) {
+      const pid_t pid = Process_getPid(curProcess);
+      const pid_t parentPid = Process_getGroupOrParent(curProcess);
+      if (parentPid == pid) {
+         looping = false;
+
+         break;
+      }
+      Process* parentProc = Hashtable_get(pt->super.table, parentPid);
+      if (parentProc == NULL) {
+         // TODO: or assert ?
+         looping = false;
+         break;
+      }
```

Comment:
```suggestion
      if (!parentProc) {
         break;
      }
```

An assert may trigger accidently here, if the parent of a process is not captured in the process scan (e.g. just started)